diesel/src/sqlite/connection/stmt.rs

```diff
@@ -48,11 +48,14 @@ impl Statement {
             )
         };
 
-        ensure_sqlite_ok(prepare_result, raw_connection.internal_connection.as_ptr()).map(|_| {
-            Statement {
-                inner_statement: unsafe { NonNull::new_unchecked(stmt) },
-            }
-        })
+        ensure_sqlite_ok(prepare_result, raw_connection.internal_connection.as_ptr())?;
+
+        // sqlite3_prepare_v3 returns a null pointer for empty statements. This includes
+        // empty or only whitespace strings or any other non-op query string like a comment
+        let inner_statement = NonNull::new(stmt).ok_or_else(|| {
+            crate::result::Error::QueryBuilderError(Box::new(crate::result::EmptyQuery))
```

Comment:
Probably that not the `QueryBuilder` error? Query was built successfully, the error at prepare phase.

Reply:
Preparing the statement is the last step of building your query, so that's the most fitting error kind. We also use this erorr variant in similar situations, for example for update statements that won't update any column at all.